### PR TITLE
feat: voice chat in join room page

### PR DESCRIPTION
**Pull request by noonoo169**

### CODE REVIEW

**Review by nguyenquan23 — APPROVED** [review]

File: src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+    @OnEvent("disconnect all")
+    public void onDisconnectAll(SocketIOClient client){
+        String roomId = rooms.get(client.getSessionId().toString());
+        for(String sessionId : usersIdRooms.get(roomId)){
+            server.getClient(UUID.fromString(sessionId)).sendEvent("disconnectRoom");
+//            server.getRoomOperations(roomId).sendEvent("user left", sessionId);
+//            users.remove(sessionId);
+//            rooms.remove(sessionId);
+//            usersIdRooms.get(roomId).remove(sessionId);
```
Comment (ces-khanhhoang) [inline src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java]: Remove unused code

File: src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        }
+    }
+
+    @OnEvent("join room")
+    public void onJoinRoom(SocketIOClient client, String roomId){
+//        if(usersIdRooms.containsKey(roomId)){
+//            usersIdRooms.remove(roomId);
+//        }
```
Comment (ces-khanhhoang) [inline src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java]: as above

File: src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (ces-khanhhoang) [inline src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java]: Remove this white space

File: src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java (hunk 1 above)
Comment (noonoo169) [reply src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java]: fixed

File: src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java (hunk 2 above)
Comment (noonoo169) [reply src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java]: fixed

File: src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java (hunk 3 above)
Comment (noonoo169) [reply src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java]: fixed

File: src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+import java.util.*;
+
+@Component
+@CrossOrigin("*")
+@Slf4j
+public class SocketHandler {
+    private final SocketIOServer server;
+    private static final Map<String, SocketIOClient> users = new HashMap<>(); // <SessionId, client>
+    private static final Map<String, String> rooms = new HashMap<>(); // <SessionId, RoomId>
+    private static final Map<String, List<String>> usersIdRooms = new HashMap<>(); // <idRoom, SessionId>
```
Comment (ces-khanhhoang) [inline src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java]: remove unused code

File: src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (ces-khanhhoang) [inline src/main/java/com/example/thetelephoneappbe/voice/socket/SocketHandler.java]: remove white space

**Comment by ces-khanhhoang** [comment]
Name commit wrong should be set as feat/add-voice-chat

**Review by ces-khanhhoang — APPROVED** [review]

**Pull request merged**